cookbooks/clearwater/recipes/local_config.rb

```diff
@@ -42,22 +42,41 @@
 
 # Set up the local config file
 
-# Determine GR site names
-if node[:clearwater][:gr]
-  if node[:clearwater][:index] and node[:clearwater][:index] % 2 == 1
-    local_site = "odd_numbers"
-    remote_site = "even_numbers"
-  else
-    local_site = "even_numbers"
-    remote_site = "odd_numbers"
+nodes = search(:node, "chef_environment:#{node.chef_environment}")
+
+# Check if we have a GR deployment, and setup the correct configuration if we do.
+if node[:clearwater][:gr] && node[:clearwater][:gr] > 1 && node[:clearwater][:index]
+  number_of_sites = node[:clearwater][:gr]
+
+  # Set up an array of all the sites.
+  sites = Array.new(number_of_sites)
+  for i in 0...number_of_sites
+      sites[i] = "site#{i}"
+  end
+
+  # Work out which site this node is in based on its index.
+  local_site_index = node[:clearwater][:index] % number_of_sites
+  local_site = sites[local_site_index]
+
+  # Remove the local site to get the list of remote sites.
+  sites.delete_at(local_site_index)
+  remote_sites = sites.join(",")
+
+  # List all nodes in the remote sites as the remote_cassandra_nodes
```

Comment:
I think (???) that this means that your Sprout/Ralf/Ellis nodes will have a remote_cassandra_nodes setting, set to the value of all the other nodes of that type. I can see the value in that - it means we don't need to specify Homestead, Homer, Memento etc. explicitly - but worth a comment to be clear that that's what's happening.